src/mlpack/methods/ann/init_rules/gaussian_init.hpp

```diff
@@ -49,13 +49,26 @@ class GaussianInitialization
    */
   void Initialize(arma::mat& W,
                   const size_t rows,
-                  const size_t cols)
+                  const size_t cols,
+                  bool truncated = false)
```

Comment:
The issue I see here, is that the user can't pass ``truncated = true``, if the initialization rule is used inside the neural network codebase. So, I guess a simple solution would be to write an extra class. Let me know what you think.

Reply:
Yeah, you're right  - I've created a new class `TruncatedGaussianInitialization` now. A couple of points I'd like your thoughts on:

1. Would it be better to have a separate file `truncated_gaussian_init.hpp`, as opposed to placing both `GaussianInitialization` and `TruncatedGaussianInitialization` under the same roof? I don't see any file having multiple class definitions anywhere else in the codebase, so let me know if you think its better to pull `TruncatedGaussianInitialization` out into a new file.

2. I've let the `truncated` flag in the `Initialize` methods of `GaussianInitialization` be as it is; and used a `GaussianInitialization` object in the `TruncatedGaussianInitialization` class. I know its generally better to have loosely coupled class definitions, but I think these two are similar enough for us to get away with this?

Reply:
Another idea would be to provide the flag as a template parameter, i.e., make a class `GaussianInitializationType<bool>`, then

```
/**
 * documentation for GaussianInitiazliation
 */
typedef GaussianInitializationType<false> GaussianInitialization;

/**
 * documentation for TruncatedGaussianInitialization
 */
typedef GaussianInitializationType<true> TruncatedGaussianInitialization;
```

Maybe that might help reduce the amount of code.  I am not picky though, I think it's fine to do it either way (two separate classes or one with a template parameter and typedefs).  I just thought it might be helpful to suggest it. :)

Reply:
Agreed, that is also an option, @PlantsAndBuildings your decision.